File: Zad4 /Strategia.py
class BicycleDeliveryStrategy:
    def deliver(self):
        print("Delivered by bicycle!")


class PublicTransportDeliveryStrategy:
    def deliver(self):
        print("Delivered by public transport!")


class TaxiDeliveryStrategy:
    def deliver(self):
        print("Delivered by taxi!")


class DeliverContext:
    def __init__(self, deliveryStrategy):
        self.deliveryStrategy = deliveryStrategy

    def DeliverProduct(self):
        self.deliveryStrategy.deliver()


money = input("Type in money available: ")
time = input("Type in time available: ")
value = input("Type in product value: ")

if money >= 0 and money < 3:
    context = DeliverContext(BicycleDeliveryStrategy())
elif money >= 3 and money < 20:
    context = DeliverContext(BicycleDeliveryStrategy(
    )) if value < 0.25 and time > 30 else DeliverContext(PublicTransportDeliveryStrategy())
elif money >= 20:
    if value > 0.75:
        context = DeliverContext(TaxiDeliveryStrategy())
    elif value >= 0.25:
        if time > 30:
            context = DeliverContext(BicycleDeliveryStrategy())
        else:
            context = DeliverContext(PublicTransportDeliveryStrategy())
    else:
        context = DeliverContext(BicycleDeliveryStrategy(
        )) if value < 0.25 and time > 30 else DeliverContext(PublicTransportDeliveryStrategy())


context.DeliverProduct()

File: Zad4 /Strategia.cs
using System;

namespace Zad4
{
    class Program
    {
        static void Main(string[] args)
        {
            Decimal money;
            double time;
            double value;

            DeliverContext context = null;

            System.Console.WriteLine("Type in money available: ");
            money = Convert.ToDecimal(Console.ReadLine());
            System.Console.WriteLine("Type in time available: ");
            time = Convert.ToDouble(Console.ReadLine());
            System.Console.WriteLine("Type in product value: ");
            value = Convert.ToDouble(Console.ReadLine());

            switch (money)
            {
                case Decimal d when (money >= 0 && money < 3):
                    context = new DeliverContext(new BicycleDeliveryStrategy());
                    break;
                case Decimal d when (money >= 3 && money < 20):
                    context = value < 0.25 && time > 30 ? new DeliverContext(new BicycleDeliveryStrategy()) : new DeliverContext(new PublicTransportDeliveryStrategy());
                    break;
                case Decimal d when (money >= 20):
                    if (value > 0.75)
                    {
                        context = new DeliverContext(new TaxiDeliveryStrategy());
                    }
                    else if (value >= 0.25)
                    {
                        if (time > 30)
                        {
                            context = new DeliverContext(new BicycleDeliveryStrategy());
                        }
                        else
                        {
                            context = new DeliverContext(new PublicTransportDeliveryStrategy());
                        }
                    }
                    else
                    {
                        context = value < 0.25 && time > 30 ? new DeliverContext(new BicycleDeliveryStrategy()) : new DeliverContext(new PublicTransportDeliveryStrategy());
                    }
                    break;
            }

            context.DeliverProduct();
        }
    }
}
